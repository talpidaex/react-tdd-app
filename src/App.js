import "./App.css";

function App() {
  return (
    <div>
      <h2>Loot Check</h2>
    </div>
  );
}

export default App;
