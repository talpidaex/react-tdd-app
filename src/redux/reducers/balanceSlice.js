import { createSlice } from "@reduxjs/toolkit";

const balanceSlice = createSlice({
  name: "balance",
  initialState: {
    balance: 0,
  },
  reducers: {
    balanceAdded: (state, action) => {
      state.balance = action.payload;
    },
  },
  extraReducers: (builder) => {},
});
export const { balanceAdded } = balanceSlice.actions;

export default balanceSlice.reducer;
