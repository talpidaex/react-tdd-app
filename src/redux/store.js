import { combineReducers, configureStore } from "@reduxjs/toolkit";
import balanceSlice from "./reducers/balanceSlice";

const rootReducer = combineReducers({
  balance: balanceSlice,
});

export function setupStore(preloadedState) {
  return configureStore({
    reducer: rootReducer,
    preloadedState,
  });
}
