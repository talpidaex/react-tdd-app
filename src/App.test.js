import { render } from "@testing-library/react";
import App from "./App";
import { renderWithProviders } from "./redux/utils/test-util";
import { setupStore } from "./redux/store";
import { balanceAdded } from "./redux/reducers/balanceSlice";

describe("App", () => {
  const { container } = render(<App />);

  it("render properly", () => {
    expect(container).toMatchSnapshot();
  });

  test("Uses preloaded state to render", () => {
    const store = setupStore();
    store.dispatch(balanceAdded(1000));
    renderWithProviders(<App />, { store });
  });
});
