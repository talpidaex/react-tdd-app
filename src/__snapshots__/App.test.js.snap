// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render properly 1`] = `
<div>
  <div>
    <h2>
      Loot Check
    </h2>
  </div>
</div>
`;
